tests/lvmo/test_lvm_snapshot_node_reboot.py

```diff
@@ -0,0 +1,308 @@
+from ocs_ci.framework.testlib import skipif_ocs_version, ManageTest
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.cluster import LVM
+from ocs_ci.ocs.exceptions import (
+    LvSizeWrong,
+    CommandFailed,
+    PodDidNotReachRunningState,
+)
+from ocs_ci.ocs.node import wait_for_nodes_status
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
+from ocs_ci.ocs.ocp import switch_to_project
+from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.node import get_node_objs
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@skipif_lvm_not_installed
+@skipif_ocs_version("<4.11")
+class TestLvmSnapshotNodeReboot(ManageTest):
+    """
+    Test lvm snapshot with node reboot
+
+    """
+
+    @pytest.mark.parametrize(
+        argnames=["volume_mode", "volume_binding_mode"],
+        argvalues=[
+            pytest.param(
+                *[constants.VOLUME_MODE_FILESYSTEM, constants.WFFC_VOLUMEBINDINGMODE],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+            pytest.param(
+                *[constants.VOLUME_MODE_BLOCK, constants.WFFC_VOLUMEBINDINGMODE],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+            pytest.param(
+                *[
+                    constants.VOLUME_MODE_FILESYSTEM,
+                    constants.IMMEDIATE_VOLUMEBINDINGMODE,
+                ],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+            pytest.param(
+                *[constants.VOLUME_MODE_BLOCK, constants.IMMEDIATE_VOLUMEBINDINGMODE],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+        ],
+    )
+    def test_create_snapshot_with_node_reboot(
+        self,
+        nodes,
+        volume_mode,
+        volume_binding_mode,
+        project_factory,
+        lvm_storageclass_factory,
+        snapshot_factory,
+        snapshot_restore_factory,
+        pvc_factory,
+        pod_factory,
+    ):
+        """
+        test create snapshot reboot
+        .* Create PVC
+        .* Create POD
+        .* Run IO PVC
+        .* Create Snapshot
+        .* Create pvc from Snapshot
+        .* Attach pod
+        .* Reboot node
+        .* Check LV size
+        .* Run IO
+        .* Create another restore
+        .* Attach Pod and check utilization
+
+        """
+
+        access_mode = constants.ACCESS_MODE_RWO
+
+        lvm = LVM(fstrim=True, fail_on_thin_pool_not_empty=False)
+        disk1 = lvm.pv_data["pv_list"][0]
+        disk_size = lvm.pv_data[disk1]["pv_size"]
+        pvc_size = int(float(disk_size)) * 2
+        thin_pool_size = lvm.get_thin_pool1_size()
+        first_io_ratio = 0.1
+        second_io_ratio = 0.05
+
+        def calc_fio_size(io_ratio, pvc__size):
+            return (
+                int(str(int(float(pvc__size)) * io_ratio).split(".")[0])
+                if "." in str(int(float(pvc__size)) * io_ratio)
+                else int(float(pvc__size) * io_ratio)
+            )
```

Comment:
this function should be available for other use cases so should be taken out of the test file

Reply:
My mistake is to do it in a complex way. I should only multiple the ratio and then use round so I don't need any function. I'll convert it to a simple phrase and will drop the function.

Reply:
Changed to a simple phrase.